### ISSUE TITLE
Docker input plugin causes Telegraf to crash 

### ISSUE THREAD

**Issue by StephanZaat** [influxdata/telegraf#1132]
I just switched from collectd to enable easier Docker metrics and I have this issue happening on several identical Docker servers at random intervals:

Everything runs great for several days and then the metrics are gone due to the following error:

...
```
2016/04/29 09:35:51 Gathered metrics, (10s interval), from 10 inputs in 1.570824848s
2016/04/29 09:36:00 Wrote 130 metrics to output influxdb in 11.021094ms
2016/04/29 09:36:04 Error decoding: EOF
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x583e6c]

goroutine 815998 [running]:
panic(0x10241a0, 0xc820010080)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/influxdata/telegraf/plugins/inputs/docker.gatherContainerStats(0x0, 0x7f2114cc0520, 0xc8205f2b00, 0xc8207e64e0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:238 +0x352c
github.com/influxdata/telegraf/plugins/inputs/docker.(*Docker).gatherContainer(0xc820160840, 0xc820404480, 0x40, 0xc820404500, 0x1, 0x4, 0xc820404540, 0x35, 0xc820212320, 0x47, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:228 +0x963
github.com/influxdata/telegraf/plugins/inputs/docker.(*Docker).Gather.func1(0xc820558c60, 0xc820160840, 0x7f2114cc0520, 0xc8205f2b00, 0xc820404480, 0x40, 0xc820404500, 0x1, 0x4, 0xc820404540, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:112 +0x9e
created by github.com/influxdata/telegraf/plugins/inputs/docker.(*Docker).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:116 +0x58d
```

The servers that run Telegraf without the Docker plugin run very solid and I'm very happy!

**Comment by zstyblik** [comment]
@StephanZaat which version of Telegraf do you use?

**Comment by StephanZaat** [comment]
Sorry. This is the latest Telegraf - Version 0.12.1

**Comment by zstyblik** [comment]
@StephanZaat looking at the code, I believe this is/will be fixed in 0.13.x. It has been fixed by https://github.com/influxdata/telegraf/issues/1052.

**Comment by StephanZaat** [comment]
OK, that's great. I guess I should have searched a bit harder.

**Comment by sparrc** [comment]
this is https://github.com/influxdata/telegraf/issues/1052, it's been fixed in master and will be in release 0.13.0
